src/basic-samples/GuaranteedPublisher.js

```diff
@@ -0,0 +1,196 @@
+        console.log(timestamp + line);
+    };
+
+    publisher.log('\n*** publisher to topic "' + publisher.topicName + '/*" is ready to connect ***');
+
+    // main function
+    publisher.run = function (argv) {
+        publisher.connect(argv);
+    };
+
+    // Establishes connection to Solace PubSub+ Event Broker
+    publisher.connect = function (argv) {
+        if (publisher.session !== null) {
+            publisher.log('Already connected and ready to publish messages.');
+            return;
+        }
+        // extract params
+        if (argv.length < (2 + 3)) { // expecting 3 real arguments
+            publisher.log('Cannot connect: expecting all arguments' +
+                ' <protocol://host[:port]> <client-username>@<message-vpn> <client-password>.\n' +
+                'Available protocols are ws://, wss://, http://, https://, tcp://, tcps://');
+            process.exit();
+        }
+        var hosturl = argv.slice(2)[0];
+        publisher.log('Connecting to Solace PubSub+ Event Broker using url: ' + hosturl);
+        var usernamevpn = argv.slice(3)[0];
+        var username = usernamevpn.split('@')[0];
+        publisher.log('Client username: ' + username);
+        var vpn = usernamevpn.split('@')[1];
+        publisher.log('Solace PubSub+ Event Broker VPN name: ' + vpn);
+        var pass = argv.slice(4)[0];
+
+        // create session
+        try {
+            publisher.session = solace.SolclientFactory.createSession({
+                // solace.SessionProperties
+                url:      hosturl,
+                vpnName:  vpn,
+                userName: username,
+                password: pass,
+                publisherProperties: {
+                  acknowledgeMode: solace.MessagePublisherAcknowledgeMode.PER_MESSAGE,
+              }          
+            });
+        } catch (error) {
+            publisher.log(error.toString());
+        }
+        // define session event listeners
+        publisher.session.on(solace.SessionEventCode.UP_NOTICE, function (sessionEvent) {
+            publisher.log('=== Successfully connected and ready to publish messages. ===');
+            publisher.publish();
+        });
+        publisher.session.on(solace.SessionEventCode.CONNECT_FAILED_ERROR, function (sessionEvent) {
+            publisher.log('Connection failed to the message router: ' + sessionEvent.infoStr +
+                ' - check correct parameter values and connectivity!');
+        });
+        publisher.session.on(solace.SessionEventCode.DISCONNECTED, function (sessionEvent) {
+            publisher.log('Disconnected.');
+            if (publisher.session !== null) {
+                publisher.session.dispose();
+                publisher.session = null;
+            }
+        });
+        publisher.session.on(solace.SessionEventCode.ACKNOWLEDGED_MESSAGE, function (sessionEvent) {
+            publisher.log('Delivery of message with correlation key = ' +
```

Comment:
Technically this is "Delivery of message **to PubSub+ Broker** with correlation". I propose explicitly adding the broker in the context of the output message to avoid confusion of confirmed deliver to client vs broker

Reply:
Done